### DISCUSSION TITLE
Formatter rule with autofix for `ruff check`

### DISCUSSION THREAD

**Discussion by rassie** [astral-sh/ruff#13073] [Q&A]
I'm pretty sure there must have been some discussion about this, but my search-fu fails me right now.

When I'm in the frontend world, I'm using `eslint-plugin-prettier` to mark up wrong formatting and can use autofixes alongside auto-save hooks in the editor to reformat the code. Since `ruff` now combines linter and formatter, I would have assumed that an auto-fixable format-checking rule for the linter would be a welcome addition and in a way low-hanging fruit in terms of implementation. 

Is there such a rule already? If not, is there a deeper reason for that other than "nobody has implemented it yet"?

**Comment by MichaReiser** [comment]
Hi @rassie 

We do plan to integrate `format` into `check` long-term. You can read more about it in https://github.com/astral-sh/ruff/issues/8232

For editors: What I do is to enable format on save. This way my fails are always properly formatted.
